### ISSUE TITLE
r10k continuously updating elastic_stack

### ISSUE THREAD

**Issue by jadestorm** [elastic/puppet-elastic-stack#19]
Hi folk!  I have been tracking a bizarre issue with the elastic_stack module for a bit now.  I run r10k every 5 minutes currently to update our environments, but I'd been noticing that elastic_stack::repo keeps not being located on catalog runs.  Turns out it's because every 5 minutes the module is deleted and reinstalled via r10k.  This does not happen with any other modules, but I also have no idea at all why it's happening.  I wanted to bounce it off of y'all to see if you have any ideas.  I looked around your code and I don't see a single thing that explains it.  I have quite a few modules enabled (somewhere between 200 and 300), but here is a snippit with the elastic ones:

```
mod 'elastic/kibana', '6.3.0'
mod 'elastic/elastic_stack', '6.2.4'
mod 'elastic/elasticsearch', '6.3.3'
mod 'elastic/logstash', '6.1.4'
```

elastic_stack is the only one that continuously updates despite pointing at a specific version.  For reference this is:

- Puppet 5.5.4
- Puppet Server 5.3.4
- R10K 2.6.4

I switched to pointing directly at your git repository and that has resolved the issue for now.

**Comment by matt-cahill** [comment]
Is this because the forge package has the `.git` config included in it?
My setup is trying to initialise it as a git submodule because of this and leaving the deployed directory empty on my puppet servers.

**Comment by matt-cahill** [comment]
I'm not using r10k on my production system just a monolithic Git repo, but I can test if I get the same result as you with r10k. 
What r10k command do you use to deploy?

**Comment by jadestorm** [comment]
I use: r10k deploy environment -pv 
(with an appropriately set up control repo and Puppetfile -- most important part of the Puppetfile being:
mod 'elastic/elastic_stack', '6.2.4'

**Comment by jadestorm** [comment]
Oh yeah -- and an r10k.yaml of:
```
---
:postrun: ["/usr/local/sbin/r10k_postrun.sh"]
:cachedir: /opt/puppetlabs/puppet/cache/r10k
:sources:
  chass:
    basedir: /etc/puppetlabs/code/environments
    prefix: true
    remote: MY_GIT_REPO

:deploy:
  purge_whitelist:
  - ".resource_types"
  - ".resource_types/**"

:git:
  private_key: "/etc/pki/git/github"
  provider: shellgit
  username: MY_GIT_USERNAME
```

**Comment by antaflos** [comment]
This is definitely because the release tarball for 6.2.4 contains a local `.git` directory, as @matt-cahill noticed. This confuses r10k, especially when running `r10k deploy environment -p`.

**Comment by jadestorm** [comment]
Can confirm this is fixed in 6.3.1.  Closing issue.
